bagua/torch_api/contrib/cache_dataset.py

```diff
@@ -0,0 +1,62 @@
+from torch.utils.data.dataset import Dataset
+from .cache_loader import CacheLoader
+
+__all__ = ["CacheDataset"]
+
+
+class CacheDataset(Dataset):
+    def __init__(
+        self,
+        dataset: Dataset,
+        backend: str = "redis",
+        key_prefix: str = "",
+        batch_writes: int = 20,
+        **kwargs,
+    ):
+        """
+        A dataset wrapper which caches `dataset` samples.
+
+        This is useful in scenarios when `dataset` is slow to fetch a sample.
+
+        Args:
+            dataset: Dataset used for caching.
+            backend(str): The backend to use. Currently ``"redis"`` is supported.
+            key_prefix(str): Prefix of the cache key. Default ``""``.
+            batch_writes(int): How many key-value pairs written to cache once. Default ``20``, If `batch_writes > 1`, the
+                cache will delay writing non-existed key-value pairs until `batch_writes` key-value pairs are accumulated.
+                Thus it could combine multiple `set` operations into one `mset` operation, and is expected to reduce
+                the write latency.
+
+        Example::
+
+            >>> from bagua.torch_api.contrib import CacheDataset
+            >>> cache_dataset = CacheDataset(
+            ...     dataset, backend="redis", hosts=None, cluster_mode=False
+            ... )
+            >>> dataloader = torch.utils.data.DataLoader(cached_dataset)
+
+        .. note::
+            `CacheDataset` is a special use case of `CacheLoader`, and parameter `backend`, `key_prefix` and `batch_writes`
+            in `CacheDataset` have the same meanings with those in `CacheLoader`. See :class:`bagua.torch_api.contrib.CacheLoader`
+            for more information.
+
+        .. note::
+            The cache associates dataset indices to determined dataset samples, thus it will break the randomness of
```

Comment:
This is implied by the documentation above. And an example should be in the tutorial to illustrate real world use case, rather than here in the API doc